src/node.js

```diff
@@ -67,84 +67,96 @@
       var d = NativeModule.require('_debug_agent');
       d.start();
 
-    } else if (process._eval != null) {
-      // User passed '-e' or '--eval' arguments to Node.
-      evalScript('[eval]');
-    } else if (process.argv[1]) {
-      // make process.argv[1] into a full path
-      var path = NativeModule.require('path');
-      process.argv[1] = path.resolve(process.argv[1]);
-
-      // If this is a worker in cluster mode, start up the communication
-      // channel.
-      if (process.env.NODE_UNIQUE_ID) {
-        var cluster = NativeModule.require('cluster');
-        cluster._setupWorker();
-
-        // Make sure it's not accidentally inherited by child processes.
-        delete process.env.NODE_UNIQUE_ID;
+    } else {
+      // There is user code to be run
+
+      // Load any preload modules
+      if (process._preload_modules) {
+        var Module = NativeModule.require('module');
+        var path = NativeModule.require('path');
+        process._preload_modules.split(path.delimiter)
+          .forEach(Module._load);
```

Comment:
This will pass the name, index, and array as 3 arguments to `Module._load`, won't it? I'm pretty sure that's not what we want.